### ISSUE TITLE
Consistent Segfaults on startup

### ISSUE THREAD

**Issue by jayson** [fluent/fluent-bit#1933]
## Bug Report

**Describe the bug**
Fluent-bit segfaults shortly after start up. I was running version 1.0.3 and encountering this. I upgraded to 1.3.4 but this did not resolve the issue. I've noticed that if I drain the kubernetes node so it has no workloads on it, it will restart and work but when the node is full (100+ pods) this starts happening again.

**To Reproduce**
- logs:
```
[2020/02/05 17:47:22] [ info] [storage] initializing...
[2020/02/05 17:47:22] [ info] [storage] in-memory
[2020/02/05 17:47:22] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2020/02/05 17:47:22] [ info] [engine] started (pid=1)
[engine] caught signal (SIGTERM)
[engine] caught signal (SIGSEGV)
[2020/02/05 17:47:50] [ info] [input] pausing tail.0
#0  0x55f42d46fe00      in  in_tail_pause() at plugins/in_tail/tail.c:328
#1  0x55f42d4400a3      in  flb_input_pause_all() at src/flb_input.c:696
#2  0x55f42d448bc3      in  flb_engine_exit() at src/flb_engine.c:593
#3  0x55f42d3b6c8d      in  flb_signal_handler() at src/fluent-bit.c:242
#4  0x7fee4574905f      in  ???() at ???:0
#5  0x7fee46f2722d      in  ???() at ???:0
#6  0x55f42d5ee598      in  seekAndRead() at lib/sqlite-amalgamation-3240000/sqlite3.c:34876
#7  0x55f42d5ee6f8      in  unixRead() at lib/sqlite-amalgamation-3240000/sqlite3.c:34940
#8  0x55f42d5e49b0      in  sqlite3OsRead() at lib/sqlite-amalgamation-3240000/sqlite3.c:21635
#9  0x55f42d5f6fbe      in  readDbPage() at lib/sqlite-amalgamation-3240000/sqlite3.c:52584
#10 0x55f42d5f9a3f      in  getPageNormal() at lib/sqlite-amalgamation-3240000/sqlite3.c:55070
#11 0x55f42d5f9cd3      in  sqlite3PagerGet() at lib/sqlite-amalgamation-3240000/sqlite3.c:55185
#12 0x55f42d602920      in  getAndInitPage() at lib/sqlite-amalgamation-3240000/sqlite3.c:64141
#13 0x55f42d6063df      in  moveToChild() at lib/sqlite-amalgamation-3240000/sqlite3.c:66987
#14 0x55f42d607185      in  btreeNext() at lib/sqlite-amalgamation-3240000/sqlite3.c:67629
#15 0x55f42d6072d4      in  sqlite3BtreeNext() at lib/sqlite-amalgamation-3240000/sqlite3.c:67665
#16 0x55f42d60720e      in  btreeNext() at lib/sqlite-amalgamation-3240000/sqlite3.c:67642
#17 0x55f42d6072d4      in  sqlite3BtreeNext() at lib/sqlite-amalgamation-3240000/sqlite3.c:67665
#18 0x55f42d62228a      in  sqlite3VdbeExec() at lib/sqlite-amalgamation-3240000/sqlite3.c:87063
#19 0x55f42d619c7e      in  sqlite3Step() at lib/sqlite-amalgamation-3240000/sqlite3.c:80302
#20 0x55f42d619ecc      in  sqlite3_step() at lib/sqlite-amalgamation-3240000/sqlite3.c:80365
#21 0x55f42d650291      in  sqlite3_exec() at lib/sqlite-amalgamation-3240000/sqlite3.c:115729
#22 0x55f42d4c7a46      in  flb_sqldb_query() at src/flb_sqldb.c:120
#23 0x55f42d477996      in  flb_tail_db_file_set() at plugins/in_tail/tail_db.c:114
#24 0x55f42d47399b      in  flb_tail_file_append() at plugins/in_tail/tail_file.c:749
#25 0x55f42d475dc0      in  flb_tail_scan() at plugins/in_tail/tail_scan_glob.c:262
#26 0x55f42d46fad6      in  in_tail_init() at plugins/in_tail/tail.c:211
#27 0x55f42d43f74d      in  flb_input_instance_init() at src/flb_input.c:358
#28 0x55f42d43f82a      in  flb_input_initialize_all() at src/flb_input.c:394
#29 0x55f42d4484a1      in  flb_engine_start() at src/flb_engine.c:406
#30 0x55f42d3b7e26      in  main() at src/fluent-bit.c:853
#31 0x7fee457362e0      in  ???() at ???:0
#32 0x55f42d3b6309      in  ???() at ???:0
#33 0xffffffffffffffff  in  ???() at ???:0
```

**Expected behavior**
I don't expect segfaults at all. Perhaps an error message if something has gone wrong

**Your Environment**

* Version used: 1.0.3 and 1.3.4
* Configuration:
* Environment name and version (e.g. Kubernetes? What version?): Kubernetes 1.13.10 and 1.14.10
* Server type and version: On premises hardware
* Operating System and version: Centos 7
* Filters and plugins: None

**Additional context**
This is causing our clusters to no longer ship logs. We have started a cluster upgrade from 1.13 to 1.14 but noticed this was happening in both versions.

**Comment by jayson** [comment]
I have some more info for this. This is caused by the input.db growing in size over time and never being pruned. I looked at our input db for /var/log/containers on our kube nodes and they had 360k entries for the `in_tail_files` table. Truncating this table fixed the startup issue. Is there any plan to clean this database up automatically?

**Comment by edsiper** [comment]
@jayson 

sorry to hear about that problem.

I am releasing v1.3.7 today/tomorrow that have some improvements on that area, specifically, that rotated files are removed from the database. 

In the other side, I am interested in troubleshoot the crash issue, would be possible if I can get a copy of that sqlite database ?

**Comment by edsiper** [comment]
@jayson I've just released v1.3.7 with a couple of fixes that might be involved in the problem you have exposed. Would you please upgrade to that version and provide some feedback ?

If the issue continues I am happy to join a call so we can troubleshoot the problem.
